### ISSUE TITLE
Add support for `DelayStamp` to the AzureTransport class to delay message retries after failures.

### ISSUE THREAD

**Issue by akserikawa** [AymDev/MessengerAzureBundle#6]
Symfony Messenger allows to configure the time delay in milliseconds after which a failed message can be retried.

https://symfony.com/doc/current/messenger.html#retries-failures

Scrolling through the `AzureTransport` class, I can see it sends a message as soon as it's passed to `send(Envelope $envelope)`. So even though Symfony adds the DelayStamp to the Envelope after it fails, the AzureTransport ignores it completely. 

Is there a way to add support for this?
What could be a viable workaround meanwhile?

Thanks!

**Comment by akserikawa** [comment]
Spent a bit of time on a workaround. In my own Serializer implementation, I override the `BrokerProperties` HTTP header from the `AzureTransport` class to set the `ScheduledEnqueueTiemUtc` property with the correct delay for each message once they failed at least once. 

```
/**
     * As we don't have access to any of the vendor code in AymDev\MessengerAzureBundle,
     * we need to override the `BrokerProperties.ScheduledEnqueueTimeUtc` property to be able to postpone the processing
     * of a message that has failed at least once already, applying the time delay configured in DelayStamp.
     * Once the `ScheduledEnqueueTimeUtc` is reached, the message will be processed again immediately.
     * @akira.serikawa
     */
    private function overrideBrokerPropertiesStampToReScheduleMessage(Envelope $envelope, array $headers): array
    {
        /** @var null|DelayStamp $delayStamp */
        if ($delayStamp = $envelope->last(DelayStamp::class)) {
            /** @var null|AzureBrokerPropertiesStamp $brokerProperties */
            if ($brokerProperties = $envelope->last(AzureBrokerPropertiesStamp::class)) {
                $brokerProperties->setScheduledEnqueueTimeUtc(
                    new DateTimeImmutable(
                        \sprintf("+%d seconds", ($delayStamp->getDelay() / 1000))
                    )
                );
                $headers['BrokerProperties'] = $brokerProperties->encode();
            }
        }

        return $headers;
    }
```

It works. But it would be great if the bundle did this internally.
Do you think this is a good idea for a PR? @AymDev If yes I'm down to work on one.

**Comment by AymDev** [comment]
Hello @akserikawa , yes I think that would be a good idea for a PR !
I did not need this feature when I created the bundle, I forgot it...

Ideally, I would merge a PR if:

 - it adds the `ScheduledEnqueueTimeUtc` property if there is a `DelayStamp` (creating the `BrokerPropertiesStamp` if needed)
 - it adds a `DelayStamp` when decoding a received message (like we already do for `TransportMessageIdStamp` [here](https://github.com/AymDev/MessengerAzureBundle/blob/main/src/Messenger/Transport/AzureTransport.php#L133-L136))
 - it tests the 2 described features
 
 Sorry again for the response delay and thanks for your contribution.

**Comment by akserikawa** [comment]
Alright that's good to hear!

I'll start working on it whenever I have time :)

**Comment by HUG-socloz** [comment]
Funny I was just working on it.
If it could work for you ? [PR-8](https://github.com/AymDev/MessengerAzureBundle/pull/8)

@AymDev 
When do you need to add DelayStamp on decoding a received message ?

**Comment by AymDev** [comment]
Thanks @HUG-socloz ! I'll check your PR.
I thought that it could be useful to know if the message was delayed but it may be a useless feature. The official transports don't do it, we can forget it.

**Comment by AymDev** [comment]
Solved by #8 
Thanks for your contributions 🙂
